conjure-lib/src/main/java/com/palantir/conjure/java/lib/DoubleArrayList.java

```diff
@@ -0,0 +1,167 @@
+        this(16);
+    }
+
+    private void resizeIfNecessary(int toAdd) {
+        if (size + toAdd > elements.length) {
+            elements = Arrays.copyOf(elements, newCapacity(size + toAdd));
+        }
+    }
+
+    @SuppressWarnings("ThrowError")
+    private int newCapacity(int minCapacity) {
+        int oldCapacity = elements.length;
+        // increase by 50%
+        int newCapacity = oldCapacity + (oldCapacity >> 1);
+        if (newCapacity - minCapacity <= 0) {
+            if (elements.length == 0) {
+                return Math.max(10, minCapacity);
+            }
+            if (minCapacity < 0) {
+                throw new OutOfMemoryError();
+            }
+            return minCapacity;
+        }
+        return (newCapacity - MAX_ARRAY_SIZE <= 0) ? newCapacity : hugeCapacity(minCapacity);
+    }
+
+    @SuppressWarnings("ThrowError")
+    private static int hugeCapacity(int minCapacity) {
+        if (minCapacity < 0) {
+            throw new OutOfMemoryError();
+        }
+        return (minCapacity > MAX_ARRAY_SIZE) ? Integer.MAX_VALUE : MAX_ARRAY_SIZE;
+    }
+
+    public DoubleArrayList(int initialCapacity) {
+        elements = new double[initialCapacity];
+    }
+
+    public DoubleArrayList(Collection<Double> collection) {
+        elements = new double[collection.size()];
+        size = collection.size();
+        if (size != 0) {
+            int idx = 0;
+            for (Double value : collection) {
+                elements[idx] = value;
+                idx++;
+            }
+        }
+    }
+
+    public DoubleArrayList(double[] doubleArray) {
+        size = doubleArray.length;
+        elements = Arrays.copyOf(doubleArray, size);
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public void add(int index, Double toAdd) {
+        checkIndexInAddRange(index);
+        resizeIfNecessary(1);
+        if (index == size) {
+            resizeIfNecessary(1);
+            size++;
+        } else {
+            resizeIfNecessary(1);
+            // Move over all entries
+            System.arraycopy(elements, index, elements, index + 1, size - index);
+            size++;
+        }
+
+        elements[index] = toAdd;
+    }
+
+    @Override
+    public boolean addAll(int index, Collection<? extends Double> collection) {
+        checkIndexInAddRange(index);
+        int numToAdd = collection.size();
+        if (numToAdd == 0) {
+            return false;
+        }
+        resizeIfNecessary(numToAdd);
+        System.arraycopy(elements, index, elements, index + numToAdd, size - index);
+        int offset = 0;
+        for (Double element : collection) {
+            elements[index + offset] = element;
+            offset++;
+        }
+        size += numToAdd;
+        return true;
+    }
+
+    @Override
+    public boolean addAll(Collection<? extends Double> collection) {
+        return addAll(size, collection);
+    }
```

Comment:
AbstractList already provides this

Reply:
```
public boolean addAll(int index, Collection<? extends E> c) {
    rangeCheckForAdd(index);
    boolean modified = false;
    for (E e : c) {
        add(index++, e);
        modified = true;
    }
    return modified;
}
```

This is the implementation I see from AbstractCollection which AbstractList uses. Better of using the DoubleArrayList implementation that only does one array resize and check.